tests/e2e/specs/api/DevfileAcceptanceTestAPI.spec.ts

```diff
@@ -3,47 +3,102 @@ import { DevWorkspaceConfigurationHelper } from '../../utils/DevWorkspaceConfigu
-        const devWorkspaceConfigurationYamlString: string = await devWorkspaceConfigurationHelper.getDevWorkspaceConfigurationYamlAsString(devfileContext);
-        const applyOutput: ShellString = kubernetesCommandLineToolsExecutor.applyYamlConfigurationAsStringOutput(devWorkspaceConfigurationYamlString);
+                expect(applyOutput.stdout)
+                    .contains('devworkspacetemplate')
+                    .and.contains('devworkspace')
+                    .and.contains.oneOf(['created', 'configured']);
 
-        expect(applyOutput.stdout)
-            .contains('devworkspacetemplate')
-            .and.contains('devworkspace')
-            .and.contains.oneOf(['created', 'configured']);
+            });
 
-    });
+            test('Wait until DevWorkspace has status "ready"', async function (): Promise<void> {
+                expect(kubernetesCommandLineToolsExecutor.waitDevWorkspace().stdout).contains('condition met');
+            });
 
-    test('Wait until DevWorkspace has status "ready"', async function (): Promise<void> {
-        expect(kubernetesCommandLineToolsExecutor.waitDevWorkspace().stdout).contains('condition met');
-    });
+            test('Check if project was created', function (): void {
+                clonedProjectName = StringUtil.getProjectNameFromGitUrl(devfileSample.link);
+                expect(containerTerminal.ls().stdout).includes(clonedProjectName);
+            });
 
-    suiteTeardown('Delete DevWorkspace', async function (): Promise<void> {
-        kubernetesCommandLineToolsExecutor.deleteDevWorkspace();
-    });
-});
+            test('Check if files were imported', function (): void {
+                containerWorkDir = containerTerminal.pwd().stdout.replace('\n', '');
+                expect(containerTerminal.ls(`${containerWorkDir}/${clonedProjectName}`).stdout).includes(`devfile.yaml`);
+            });
 
+            test(`Check if build command returns success`, function (): void {
+                if (devfilesBuildCommands.length === 0) {
+                    Logger.info(`Devfile does not contains build commands.`);
+                } else {
+                    let workingDir: string;
+                    devfilesBuildCommands.forEach((command) => {
+                        Logger.info(`command.exec.commandLine: ${command.exec.commandLine};\n
+                        command.exec.component: ${command.exec.component};\n
+                        command.exec.workingDir: ${command.exec.workingDir}.`);
+
+                        workingDir = StringUtil.getFullWorkingDirPathExplicit(command.exec.workingDir, containerTerminal);
+                        const commandString: string = `cd ${workingDir} && ${command.exec.commandLine}`;
+                        Logger.info(`Full build command to be executed: ${commandString}`);
+
+                        const output: ShellString = containerTerminal.executeCommand(commandString, command.exec.component);
```

Comment:
how does it react on long build time?
Is it possible to set command build timeout, e.g. adding prefix to command `timeout $commandTimeoutSec <command>`?

Reply:
For example build of quarkus sample from github registry takes more than 20 minutes.. I tried this approach you propose but than decided that mocha test timeout will be enough

Reply:
I see.

How can mocha test timeout be set when call the test?

Reply:
25 minutes is way too much.
Workspace idle timeout is 10 minutes, as far as I remember.

At the same time project could build quite a long time in disconnected environment when use proxied artifactory.

Let's keep it as it for build commands.

Specific timeout setting will be needed when test non-terminating commands like run and debug.